spec/integration/util/windows/principal_spec.rb

```diff
@@ -121,6 +122,28 @@
       expect(principal.to_s).to eq(builtin_localized)
     end
 
+    it "should always sanitize the account name first" do
+      expect(Puppet::Util::Windows::SID::Principal).to receive(:sanitize_account_name).with('NT AUTHORITY\\SYSTEM').and_call_original
+      Puppet::Util::Windows::SID::Principal.lookup_account_name('NT AUTHORITY\\SYSTEM')
+    end
+
+    it "should be able to create an instance from an account name prefixed by APPLICATION PACKAGE AUTHORITY" do
+      expect(Puppet::Util::Windows::SID::Principal).to receive(:sanitize_account_name).with('APPLICATION PACKAGE AUTHORITY\\ALL APPLICATION PACKAGES').and_call_original
```

Comment:
Do you need to expect on this call? Testing that calling `lookup_account_name` is returning the expected data would suffice?